File: java/src/com/db4o/dg2db4o/chapter7/ZipCode.java

package com.db4o.dg2db4o.chapter7;


public class ZipCode {
    private String _state;
    private String _code;
    private String _extension;
    
    public ZipCode(String state, String code, String extension) {
        _state = state;
        _code= code;
        _extension = extension;
    }
    
    public String getState() {
        return _state;
    }
    
    public void setState(String state) {
        this._state = state;
    }
    
    public String getCode() {
        return _code;
    }
    
    public void setCode(String code) {
        this._code = code;
    }
    
    public String getExtension() {
        return _extension;
    }
    
    public void setExtension(String extension) {
        this._extension = extension;
    }
    
    public String toString() {
        return _state + _code + "-" + _extension + " (ZipCode)";
    }
    
}

File: csharp/Chapter07/ZipCode.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace com.db4o.dg2db4o.chapter7
{
    class ZipCode
    {
        string _state;
        string _code;
        string _extension;

        public ZipCode(string state, string code, string extension)
        {
            _state = state;
            _code = code;
            _extension = extension;
        }

        public string State
        {
            get
            {
                return _state;
            }
            set
            {
                _state = value;
            }
        }

        public string Code
        {
            get
            {
                return _code;
            }
            set
            {
                _code = value;
            }
        }

        public string Extension
        {
            get
            {
                return _extension;
            }
            set
            {
                _extension = value;
            }
        }

        public override string ToString()
        {
            return _state + _code + "-" + _extension + " (ZipCode)";
        }
    }
}
